src/main/scala/com/twitter/finagle/websocket/WebSocketHandler.scala

```diff
@@ -10,11 +9,38 @@ import org.jboss.netty.handler.codec.http.websocketx._
 import org.jboss.netty.handler.codec.http.{HttpHeaders, HttpRequest, HttpResponse}
 import scala.collection.JavaConversions._
 
+private[finagle] class CancelledException extends Exception
+
 class WebSocketHandler extends SimpleChannelHandler {
   protected[this] val messagesBroker = new Broker[String]
   protected[this] val binaryMessagesBroker = new Broker[Array[Byte]]
   protected[this] val closer = new Promise[Unit]
 
+  implicit class RichChannelFuture(future: ChannelFuture) {
```

Comment:
I've fallen behind the state of Finagle's art. Why is this necessary? Is Finagle not providing something. Plus the exception will now be a finagle-websocket exception (as opposed to a Finagle cancelled exception). Is that necessary?